src/DevChatter.Bot.Web/wwwroot/js/sprite.js

```diff
@@ -1,24 +1,80 @@
-  var rand = Math.random() * 2 * Math.PI //get random val between 0 and 2 Pi ~ 0 and 360 degrees
+  var rand = Math.random() * 2 * Math.PI; //get random val between 0 and 2 Pi ~ 0 and 360 degrees
   
   return rand;
 };
 
-Sprite.prototype.update = function () {
-  var mod = 2
+Sprite.prototype.getRandomModifier = function () {
+  var rand = Math.random() * 3 + 1; //get random value between 1 and 4 (float)
+  return rand;
+};
+
+Sprite.prototype.update = function (canvas) {
   //assuming right is 0 and left is pi rad, though doesn't really matter
   //mod ~ modifier affects magnitude of displacement
-  this.x += mod * Math.cos(this.direction) //apply degree to trig function for x displacement
-  this.y += mod * Math.sin(this.direction) //apply degree to trig function for y displacement
+
+  x_right_border = canvas.width - this.image.width; //width of canvas - width of image, insures the right border of the image bounces off the right border of the canvas
+  y_bottom_border = canvas.height - this.image.height; //same as above but for bottom of image and canvas
+  x_left_border = 0;
+  y_top_border = 0;
+
+  this.x += this.mod * Math.cos(this.direction); //apply degree to trig function for x displacement
+  this.y += this.mod * Math.sin(this.direction); //apply degree to trig function for y displacement
+
+  // screen coords are defined as 0,0 in the top left and max width, height in the bottom right
+
+
+  //check if x position has moved beyond the left or right border
+  if ((this.x < x_left_border || this.x > x_right_border) && !this.bounced) {
+    // this.direction += Math.PI / 2;
+    this.direction = this.getBounceAngle(this.direction, true);
+    this.bounced = true;
+  }
+
+  //check if y position has moved beyond the upper or lower border
+  if ((this.y < y_top_border || this.y > y_bottom_border) && !this.bounced) {
+    // this.direction += Math.PI / 2;
+    this.direction = this.getBounceAngle(this.direction, false);
+    this.bounced = true;
+  }
+
+  //not sure if this could be removed and replaced with a canvas.intersects(this.x, this.y) or something simmilar. Could also be moved to a intersects() function
+  if ((this.y < y_bottom_border && this.y > y_top_border) && (this.x > x_left_border && this.x < x_right_border)) {
+    this.bounced = false; //if sprite is within the box reset bounce flag
+  }
+
+};
+
+// takes angle of colision and a bool to tell which kind of wall it's bouncing off of
+// outputs the incident angle that the sprite will be going in
+Sprite.prototype.getBounceAngle = function (angle, vertical_wall = true) { 
+  var FULL_CIRCLE = 2 * Math.PI;
+  var bounce_angle = 0;
+
+  
+  //simplified angle calc code by benrick
+  if (vertical_wall) {
+    bounce_angle = Math.PI - angle;
+  } else {
+    bounce_angle = Math.PI * 2 - angle;
+  }
+
+  //if angle has gone passed 360 or below 0, correct this. The trig functions shouldn't care though the logic above is broken if the angle isn't within 0 and 360 (0 and 2pi)
```

Comment:
Neither code did as far as I know (after I fixed angle's on my implementation). As long as it reflects properly and displacement occurs before the bounce functions than it probably isn't necessary to keep track of whether or not is is back within the boundary.

Reply:
Sorry, misunderstood your question. I think so yes, but not under normal circumstances. If you had one sprite bouncing around the screen clockwise starting at 45 degrees it would go to 135 then 225 then -45 then 405 and so on. If i've done that right. I'll try it in code just out of curiosity.